onnx/backend/test/runner/__init__.py

```diff
@@ -298,15 +298,21 @@ def run(test_self, device):  # type: (Any, Text) -> None
                     tensor = onnx.TensorProto()
                     with open(input_file, 'rb') as f:
                         tensor.ParseFromString(f.read())
-                    inputs.append(numpy_helper.to_array(tensor))
+                    if sys.byteorder == 'big':
+                        inputs.append(numpy_helper.to_array(tensor).byteswap())
+                    else:
+                        inputs.append(numpy_helper.to_array(tensor))
                 ref_outputs = []
                 ref_outputs_num = len(glob.glob(os.path.join(test_data_dir, 'output_*.pb')))
                 for i in range(ref_outputs_num):
                     output_file = os.path.join(test_data_dir, 'output_{}.pb'.format(i))
                     tensor = onnx.TensorProto()
                     with open(output_file, 'rb') as f:
                         tensor.ParseFromString(f.read())
-                    ref_outputs.append(numpy_helper.to_array(tensor))
+                    if sys.byteorder == 'big':
+                        ref_outputs.append(numpy_helper.to_array(tensor).byteswap())
```

Comment:
This changes the test runner, but not the underlying helper method.
With this change tests may pass, but the effect may be misleading.

Should the `byteswap()` call be moved into `numpy_helper.to_array` itself?

Reply:
Thanks for the comments! 
At first I thought it might be better to do it, but I was not sure if it is always ok to byteswap the array in `numpy_helper.to_array()` .